public/utils/message_content_puller.test.ts

```diff
@@ -21,21 +23,78 @@ describe('MessageContentPool', () => {
     messageContentPool.start();
     messageContentPool.inputComplete();
 
-    await waitFor(() => {
-      expect(subscriptionMock).toHaveBeenCalledTimes(3);
-      expect(subscriptionMock).toHaveBeenNthCalledWith(1, {
-        messageContent: 'a'.repeat(10),
-        messageId: 'a',
-      });
-      expect(subscriptionMock).toHaveBeenNthCalledWith(2, {
-        messageContent: 'b'.repeat(10),
-        messageId: 'b',
-      });
-      expect(subscriptionMock).toHaveBeenNthCalledWith(3, {
-        messageContent: 'a'.repeat(10),
-        messageId: 'a',
-      });
-      expect(completeSubscriptionMock).toHaveBeenCalledTimes(1);
+    await waitFor(
+      () => {
+        expect(subscriptionMock).toHaveBeenCalledTimes(3);
+        expect(subscriptionMock).toHaveBeenNthCalledWith(1, {
+          messageContent: 'a'.repeat(10),
+          messageId: 'a',
+        });
+        expect(subscriptionMock).toHaveBeenNthCalledWith(2, {
+          messageContent: 'b'.repeat(10),
+          messageId: 'b',
+        });
+        expect(subscriptionMock).toHaveBeenNthCalledWith(3, {
+          messageContent: 'a'.repeat(10),
+          messageId: 'a',
+        });
+        expect(completeSubscriptionMock).toHaveBeenCalledTimes(1);
+      },
+      {
+        timeout: 10000,
```

Comment:
Nit: can we use `jest.useFakeTimer` here?  Then these tests won't take too much time.

Reply:
Sure, updated.